src/Avalonia.Base/Media/DrawingContext.cs

```diff
@@ -400,6 +403,20 @@ public PushedState PushTransform(Matrix matrix)
             return new PushedState(this);
         }
 
+        /// <summary>
+        /// Pushes render options.
+        /// </summary>
+        /// <param name="renderOptions">The render options.</param>
+        /// <returns>A disposable to undo the render options.</returns>
+        public PushedState PushRenderOptions(RenderOptions renderOptions)
+        {
+            PushRenderOptionsCore(renderOptions);
```

Comment:
I believe we can implement the merge here.

Reply:
The DrawingContext does not know anything about current RenderOptions. Current RenderOptions are stored in the actual implementation.